central/resourcecollection/datastore/datastore_impl.go

```diff
@@ -3,18 +3,120 @@ package datastore
 import (
 	"context"
 
+	"github.com/heimdalr/dag"
 	"github.com/stackrox/rox/central/resourcecollection/datastore/index"
 	"github.com/stackrox/rox/central/resourcecollection/datastore/search"
-	"github.com/stackrox/rox/central/resourcecollection/datastore/store/postgres"
+	"github.com/stackrox/rox/central/resourcecollection/datastore/store"
 	v1 "github.com/stackrox/rox/generated/api/v1"
 	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/logging"
 	pkgSearch "github.com/stackrox/rox/pkg/search"
+	"github.com/stackrox/rox/pkg/sync"
+)
+
+var (
+	log           = logging.LoggerForModule()
+	initBatchSize = 20
 )
 
 type datastoreImpl struct {
-	storage  postgres.Store
+	storage  store.Store
 	indexer  index.Indexer
 	searcher search.Searcher
+
+	graphLock sync.Mutex
+	graph     *dag.DAG
+}
+
+func resetLocalGraph(ds *datastoreImpl) {
+
+	ds.graphLock.Lock()
+	defer ds.graphLock.Unlock()
+
+	if ds.graph != nil {
+		ds.graph = nil
+	}
+}
+
+func (ds *datastoreImpl) initGraph(ctx context.Context) error {
+	if ds.graph != nil {
+		return nil
+	}
+
+	ds.graphLock.Lock()
+	defer ds.graphLock.Unlock()
+
+	ds.graph = dag.NewDAG()
+
+	// add ids first
+	ids, err := ds.storage.GetIDs(ctx)
```

Comment:
If I understand the postgres code correctly, this will return all the ids _that can be seen by ctx_. In other words if you are logged in with a scope that can only see some collections, this won't get all collections. 

I recommend using a special context that has all access just for the init function.